homeassistant/components/aussie_broadband/config_flow.py

```diff
@@ -0,0 +1,192 @@
+            user_input[CONF_PASSWORD],
+            async_get_clientsession(self.hass),
+        )
+        try:
+            return await self.client.login()
+        except AuthenticationException:
+            return {"base": "invalid_auth"}
+        except ClientError:
+            return {"base": "cannot_connect"}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        errors = None
+        if user_input is not None:
+            result = await self.auth(user_input)
+            if result is True:
+                await self.async_set_unique_id(user_input[CONF_USERNAME])
+                self._abort_if_unique_id_configured()
+
+                self.data = user_input
+                self.services = await self.client.get_services()
+
+                if len(self.services) == 0:
+                    return self.async_abort(reason="no_services_found")
+
+                if len(self.services) == 1:
+                    return self.async_create_entry(
+                        title=self.data[CONF_USERNAME],
+                        data=self.data,
+                        options={CONF_SERVICES: [str(self.services[0][SERVICE_ID])]},
+                    )
+
+                # Account has more than one service, select service to add
+                return await self.async_step_service()
+            errors = result
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
+            errors=errors,
+        )
+
+    async def async_step_service(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the optional service selection step."""
+        if user_input is not None:
+            return self.async_create_entry(
+                title=self.data[CONF_USERNAME], data=self.data, options=user_input
+            )
+
+        service_options = {str(s[SERVICE_ID]): s["description"] for s in self.services}
+        return self.async_show_form(
+            step_id="service",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(
+                        CONF_SERVICES, default=list(service_options.keys())
+                    ): cv.multi_select(service_options)
+                }
+            ),
+        )
+
+    async def async_step_reauth(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle reauth."""
+        errors = None
+        if user_input and user_input.get(CONF_USERNAME):
+            self._reauth_username = user_input[CONF_USERNAME]
+
+        elif user_input and user_input.get(CONF_PASSWORD):
+            data = {
+                CONF_USERNAME: self._reauth_username,
```

Comment:
Its presented in the very first user_input this function receives when reauth is first initiated. Line 113 sets this value in self so it can be used again when the user enters the new password.